assets/src/edit-story/components/keyboardShortcutsMenu/shortcutMenu.js

```diff
@@ -17,153 +17,147 @@
-      })),
-    [keyboardShortcuts]
+// How much less space than the full browser height can the popup occupy?
+const EXTRA_SPACE =
+  ADMIN_TOOLBAR_HEIGHT + HEADER_HEIGHT + TOP_MARGIN - 2 * BORDER_WIDTH;
+const FlexContent = styled.div`
+  display: flex;
+  flex-direction: column;
+  max-height: calc(
+    100vh - ${EXTRA_SPACE}px - ${({ bottomOffset }) => bottomOffset}px
   );
+`;
 
-  const headerLabels = useMemo(() => headers.map((h) => h.label).join(' '), [
-    headers,
-  ]);
+const CloseContainer = styled.aside`
+  position: absolute;
+  right: 4px;
+  top: 4px;
+`;
 
-  const leftHalfCount = Math.ceil(sections.length / 2);
-  const { leftSideSections, rightSideSections } = useMemo(
-    () =>
-      sections.reduce(
-        ({ leftSideSections, rightSideSections }, section, index) => {
-          if (index < leftHalfCount) {
-            return {
-              leftSideSections: [...leftSideSections, section],
-              rightSideSections,
-            };
-          } else {
-            return {
-              leftSideSections,
-              rightSideSections: [...rightSideSections, section],
-            };
-          }
-        },
-        {
-          leftSideSections: [],
-          rightSideSections: [],
-        }
-      ),
-    [sections, leftHalfCount]
-  );
+const ScrollableContent = styled.div`
+  overflow: auto;
+  flex-shrink: 1;
+`;
 
-  const renderSection = useCallback(
-    ({ title, commands }) => (
-      <SectionWrapper key={title}>
-        <SectionHeader id={title}>{title}</SectionHeader>
-        <SectionContent role="group" aria-labelledby={title}>
-          {commands.map(({ label, shortcut }) => (
-            <Fragment key={label}>
-              <Label role="listitem">{label}</Label>
-              <ShortcutLabel
-                role="listitem"
-                keys={shortcut}
-                alignment={'left'}
-              />
-            </Fragment>
-          ))}
-        </SectionContent>
-      </SectionWrapper>
-    ),
-    []
-  );
+function ShortcutMenu({ toggleMenu }) {
+  const containerRef = useRef();
+  const closeRef = useRef();
+
+  const [bottomOffset, setBottomOffset] = useState(0);
+  useLayoutEffect(() => {
+    const node = containerRef.current;
+    if (!node) {
+      return undefined;
+    }
+    const calc = () => {
+      // Find node bottom offset from bottom of browser window
+      const nodeRect = node.getBoundingClientRect();
+      const docRect = node.ownerDocument.documentElement.getBoundingClientRect();
+      setBottomOffset(docRect.height - nodeRect.y - nodeRect.height);
+    };
+    // Call it now
+    calc();
+    // Call it in a frame
+    // Disable reason: No need to cancel this one
+    // eslint-disable-next-line @wordpress/react-no-unsafe-timeout
+    setTimeout(calc);
+    // And call it every time the window resizes
+    window.addEventListener('resize', calc);
+    return () => window.removeEventListener('resize', calc);
+  }, []);
```

Comment:
This feels like a lot of logic and bloat to introduce for some edge cases. Looks like the whole editor layout goes to shit anyway once we're below ~650px height. and as it gets larger, I think the height of the toggle button/carousel menu stays roughly static.

I'd be fine with just a rough, static estimate/approximation here instead of doing a bunch of reads on the dom and adding event listeners. I think if it's a couple px off in extreme use cases it's an excellent tradeoff for getting rid of this extra work to the main thread
```
100vh - ${EXTRA_SPACE} - ${roughInferenceOfCarouselMenuHeightPlusMargin}
```

Reply:
The meta-boxes are the issue here though. The height of the meta-boxes panel when collapsed is known, so that could be factored in, but when expanded, the meta-box container could be any height from a few pixels to taking up the entire window.

If you don't have a plugin with meta-boxes, you don't see this at all, so I understand your confusion, but if you do (e.g.  "All in One SEO" as I use to test this), you'll see the issues. I do agree it's a lot of work, but I think it's necessary here, unfortunately.

Reply:
Ahh. yeaup. wasn't considering that. Thanks for pointing out